trust-quorum/gfss/src/gf256.rs

```diff
@@ -0,0 +1,302 @@
+//! are handy.
+
+// Don't tell me what operations to use in my implementations
+#![allow(clippy::suspicious_arithmetic_impl)]
+
+use core::fmt::{self, Binary, Display, Formatter, LowerHex, UpperHex};
+use core::ops::{Add, AddAssign, Div, Mul, MulAssign, Sub};
+use rand::Rng;
+use rand::distributions::{Distribution, Standard};
+use subtle::ConstantTimeEq;
+use zeroize::Zeroize;
+
+/// An element in a finite field of prime power 2^8
+///
+/// We explicitly don't enable the equality operators to prevent ourselves from
+/// accidentally using those instead of the constant time ones.
+#[repr(transparent)]
+#[derive(Debug, Clone, Copy, Zeroize)]
+pub struct Gf256(u8);
+
+impl Gf256 {
+    pub fn new(n: u8) -> Gf256 {
+        Gf256(n)
+    }
+
+    /// Return the underlying u8.
+    pub fn into_u8(self) -> u8 {
+        self.0
+    }
+
+    /// Return the multiplicative inverse (self^-1) of self
+    ///
+    /// self * self^-1 = 1
+    ///
+    /// It turns out that self^-1 = self^254 for GF(2^8), so we calculate that
+    /// in a simple, unrolled fashion.
+    ///
+    /// This strategy was borrowed from <https://github.com/dsprenkels/sss/blob/16c3fdb175497b25eb90b966991fa7ff19fbdcfe/hazmat.c#L247-L266>
+    #[rustfmt::skip]
+    pub fn invert(&self) -> Gf256 {
+        let mut result = *self * *self;     // self^2
+        let temp_4 = result * result;       // self^4
+        result = temp_4 * temp_4;           // self^8
+        let temp_9 = result * *self;        // self^9
+        result *= result;                   // self^16;
+        result *= temp_9;                   // self^25;
+        let temp_50 = result * result;      // self^50;
+        result = temp_50 * temp_50;         // self^100;
+        result *= result;                   // self^200;
+        result = result * temp_50 * temp_4; // self^254
+
+        result
+    }
+}
+
+pub const ZERO: Gf256 = Gf256(0);
+pub const ONE: Gf256 = Gf256(1);
+
+impl Display for Gf256 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl LowerHex for Gf256 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{:02x}", self.0)
+    }
+}
+
+impl UpperHex for Gf256 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{:02X}", self.0)
+    }
+}
+
+impl Binary for Gf256 {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{:08b}", self.0)
+    }
+}
+
+impl Distribution<Gf256> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Gf256 {
+        Gf256(rng.r#gen())
+    }
+}
+
+impl ConstantTimeEq for Gf256 {
+    fn ct_eq(&self, other: &Self) -> subtle::Choice {
+        self.0.ct_eq(&other.0)
+    }
+}
+impl Add for Gf256 {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        Gf256(self.0 ^ rhs.0)
```

Comment:
I'd flag this explicitly as a fundamental operation -- `GF(2**n)` is mod 2 with respect to each of the `n` bits, so adding is the same as a bitwise xor.